org.eclipse.lsp4e/src/org/eclipse/lsp4e/outline/LSSymbolsContentProvider.java

```diff
@@ -292,12 +334,45 @@ public Object[] getElements(Object inputElement) {
 		if (this.lastError != null && symbolsModel.getElements().length == 0) {
 			return new Object[] { "An error occured, see log for details" }; //$NON-NLS-1$
 		}
-		return symbolsModel.getElements();
+		return Arrays.stream(symbolsModel.getElements())
+				.filter(element -> !hideElement(element))
+				.toArray(Object[]::new);
 	}
 
 	@Override
 	public Object[] getChildren(Object parentElement) {
-		return symbolsModel.getChildren(parentElement);
+		return Arrays.stream(symbolsModel.getChildren(parentElement))
+			.filter(element -> !hideElement(element))
+			.toArray(Object[]::new);
+	}
+
+	/**
+	 * Decide whether to hide the given element or not.
+	 * Default implementation hides elements depending on a preference
+	 * that determines which {@link SymbolKind}s
+	 * (property in {@link DocumentSymbol}, {@link DocumentSymbolWithURI}s, and {@link SymbolInformation})
+	 * are to be hidden in the outline view.
+	 * Sub-classes may override this method to add language-specific filtering.
```

Comment:
I think we we do not currently support such sub-classing. I would remove the comment and make the method private.

Reply:
Ok. I was not sure if LSSymbolContentProvider is meant to be extended. I'll make the method private and remove the comment.